spring-kafka/src/test/java/com/baeldung/partitioningstrategy/KafkaApplicationUnitTesting.java

```diff
@@ -0,0 +1,164 @@
+package com.baeldung.partitioningstrategy;
+
+import static org.junit.Assert.assertEquals;
+import java.time.Duration;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import org.apache.kafka.clients.consumer.Consumer;
+import org.apache.kafka.clients.consumer.ConsumerRecord;
+import org.apache.kafka.clients.consumer.ConsumerRecords;
+import org.apache.kafka.clients.producer.ProducerConfig;
+import org.apache.kafka.common.TopicPartition;
+import org.apache.kafka.common.serialization.StringSerializer;
+import org.junit.Assert;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.kafka.core.DefaultKafkaProducerFactory;
+import org.springframework.kafka.core.KafkaTemplate;
+import org.springframework.kafka.test.EmbeddedKafkaBroker;
+import org.springframework.kafka.test.context.EmbeddedKafka;
+import org.springframework.kafka.test.utils.KafkaTestUtils;
+
+@SpringBootTest
+@EmbeddedKafka(partitions = 3, brokerProperties = {"listeners=PLAINTEXT://localhost:9092"})
+public class KafkaApplicationUnitTesting {
+    @Autowired
+    private KafkaProducer kafkaProducer;
+
+    @Autowired
+    private KafkaMessageConsumer kafkaMessageConsumer;
+
+    @Autowired
+    private EmbeddedKafkaBroker embeddedKafkaBroker;
+
+    @Autowired
+    private Consumer<String, String> consumer;
+
+    @BeforeEach
+    public void clearMessages() {
+        kafkaMessageConsumer.clearReceivedMessages();
+    }
+
+    @Test
+    public void givenDefaultPartitioner_whenSendingMessagesWithoutKey_shouldUseStickyDistribution() throws InterruptedException {
+        kafkaProducer.send("default-topic", "message1");
+        kafkaProducer.send("default-topic","message2");
+        kafkaProducer.send("default-topic","message3");
+
+        Thread.sleep(2000);
```

Comment:
It's almost never a good idea to use Thead.sleep in a test

If you need to wait for an async condition, use something like:
https://www.baeldung.com/awaitility-testing

To wait on a condition (e.g. the number of messages received), so the test can be deterministic

Reply:
Noted with this.